statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,534 @@
+            mu = self.predict(params, exog=exog, exposure=exposure,
+                              offset=offset)[:,None]
+            return self.model_dist.pmf(counts, mu, self.trunc)
+        else:
+            raise TypeError(
+                    "argument wich == %s not handled" % which)
+
+class TruncatedPoisson(GenericTruncated):
+    """
+    Poisson Zero Inflated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    exog_infl: array
+        A reference to the zero-inflated exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, offset=None, exposure=None,
+                 truncation=0, missing='none', **kwargs):
+        super(TruncatedPoisson, self).__init__(endog, exog, offset=offset,
+                                               exposure=exposure,
+                                               truncation=truncation,
+                                               missing=missing, **kwargs)
+        self.model_main_name = Poisson
+        self.model_main = Poisson(self.endog, self.exog)
+        self.model_dist = truncatedpoisson
+        self.result = TruncatedPoissonResults
+        self.result_wrapper = TruncatedPoissonResultsWrapper
+        self.result_reg = L1TruncatedPoissonResults
+        self.result_reg_wrapper = L1TruncatedPoissonResultsWrapper
+
+class GenericHurdle(CountModel):
+    __doc__ = """
+    Generic Hurdle model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, offset=None,
+                 exposure=None, missing='none', **kwargs):
+        super(GenericHurdle, self).__init__(endog, exog, offset=offset,
+                                            exposure=exposure,
+                                            missing=missing, **kwargs)
+        self.model1 = self.model_name1(
+            np.zeros_like(self.endog, dtype=np.float64), self.exog)
+        self.model2 = self.model_name2(self.endog, self.exog)
```

Comment:
shouldn't model2, the truncated model, only get endog and exog for non-zero endog observations (y > 0)?
y = 0 should not be an observation in the support of the zero-truncated model.
or is this not imposed at this point, i.e. in the generic setting?